src/static/utils/generateForm.jsx

```diff
@@ -8,8 +11,19 @@ const generateFormItem = (form, item, initialData) => {
     const options = {
         rules
     };
-    if (initialData && id in initialData) {
-        options.initialValue = initialData[id];
+    if (initialData) {
+        if (id.includes('.')) {
+            const splitArray = id.split('.');
+            if (splitArray[0] in initialData) {
+                let init = initialData[splitArray[0]];
+                for (let data of splitArray.slice(1)) {
+                    init = init[data];
```

Comment:
This piece seems either buggy either too much complicate for what it aims to do… What's the purpose of this?

Reply:
Getting the initialData from the object we get: 
{ card_stats: { waste_reduction: 1, ... }, }
with the id field like "card_stats.waste_reduction"

Reply:
Could `get` from lodash do the job? It tipically can do this: `get(object, "card_stats.waste_reduction", defaultValue)`

See: https://lodash.com/docs/4.17.5#get